### ISSUE TITLE
[FEAT] Provide ApolloClient with IntrospectionFragmentMatcher

### ISSUE THREAD

**Issue by griest024** [graycoreio/daffodil#694]
# :bulb: Feature request

## Feature Name

Provide `ApolloClient` with `IntrospectionFragmentMatcher`.

## The Desired Behavior

Driver calls will have defined behavior.

## Your Use Case

As a developer, I would like to have guaranteed fragment matching in GraphQL calls that target union types or interfaces.

## Prior Work

Making certain GraphQL calls (in development) will throw this error:
```
You are using the simple (heuristic) fragment matcher, but your queries contain union or interface types. Apollo Client will not be able to accurately map fragments. To make this error go away, use the `IntrospectionFragmentMatcher` as described in the docs: https://www.apollographql.com/docs/react/advanced/fragments.html#fragment-matcher
```

Others:

I don't think that this causes any runtime issues at the moment 🤞  so this is low priority but I'm leaving this here for posterity. The URL in the error message 404s, but [this](https://www.apollographql.com/docs/react/data/fragments/#fragments-on-unions-and-interfaces) covers the issue.

**Comment by griest024** [comment]
As part of the process of configuring the `IntrospectionFragmentMatcher`, the GraphQL schema will have to be fetched and parsed into fragment type data. This necessarily means that this process will have to be initiated from the app-side.

I propose exposing a utility function that can be called during the app's build process that will generate the fragment type data. There is also the possibility of exposing a function that takes care of creating the `IntrospectionFragmentMatcher` and apollo instance for apps that don't need any further customization.

**Comment by damienwebdev** [comment]
Fixed and closed by #752
